### ISSUE TITLE
finetune on unlabelled dataset follow up

### ISSUE THREAD

**Issue by bwanglzu** [jina-ai/finetuner#279]
- [ ] image argumentation: in `preprocess_fn`. have default built-in argumentation methods. for image use crop+color jitter + cutout (mask), leave space for text.
- [ ] hidden projection head in tuner, this `projection_head` is different than `bottleneck_net` concept within Tailor. It add another MLP head to the embedding model and project it down to `128d` by default, only for training. Will be cut-out when creating embedding model. @bwanglzu 
- [ ] unlabelled dataset: takes document and preprocessing function to apply unlabelled DA and generate labels on the fly. @tadejsv 
- [ ] instance sampler: Take instance images (with same "root image"), e.g. if positive is 2, batch size is 68, means 34 pairs of images. The rest within the batch are negative samples. @tadejsv 
- [ ] cross entropy with temperature. @bwanglzu 
- [ ] Experiment performance, Add a tutorial on how to use it.
